main.tf

```diff
@@ -1,77 +1,70 @@
-provider "aws" {}
+provider "aws" {
+}
 
 provider "aws" {
   alias = "peer"
 }
 
 data "aws_vpc" "peer" {
-  count = "${var.create_peering_connection ? 1 : 0}"
+  count = var.create_peering_connection ? 1 : 0
 
-  provider = "aws.peer"
+  provider = aws.peer
 
-  id = "${var.peer_vpc_id}"
+  id = var.peer_vpc_id
 }
 
 resource "aws_vpc_peering_connection" "this" {
-  count = "${var.create_peering_connection ? 1 : 0}"
-
-  peer_owner_id = "${var.peer_owner_id}"
-  peer_vpc_id   = "${var.peer_vpc_id}"
-  vpc_id        = "${var.vpc_id}"
-  tags          = "${merge(var.tags, map("Name", "${var.name}-${var.vpc_cidr}<->${var.peer_alias}-${data.aws_vpc.peer.cidr_block}"))}"
+  count = var.create_peering_connection ? 1 : 0
+
+  peer_owner_id = var.peer_owner_id
+  peer_vpc_id   = var.peer_vpc_id
+  vpc_id        = var.vpc_id
+  tags = merge(
+    var.tags,
+    {
+      "Name" = "${format("%v", var.name)}-${format("%v", var.vpc_cidr)}<->${format("%v", var.peer_alias)}-${data.aws_vpc.peer[0].cidr_block}"
+    },
+  )
 }
 
 resource "aws_vpc_peering_connection_accepter" "this" {
-  count = "${var.create_peering_connection ? 1 : 0}"
+  count = var.create_peering_connection ? 1 : 0
 
-  provider = "aws.peer"
+  provider = aws.peer
 
-  vpc_peering_connection_id = "${aws_vpc_peering_connection.this.id}"
+  vpc_peering_connection_id = aws_vpc_peering_connection.this[0].id
   auto_accept               = true
 
-  tags {
-    Name = "${var.peer_alias}-${data.aws_vpc.peer.cidr_block}<->${var.name}-${var.vpc_cidr}"
+  tags = {
+    Name = "${format("%v", var.peer_alias)}-${data.aws_vpc.peer[0].cidr_block}<->${format("%v", var.name)}-${format("%v", var.vpc_cidr)}"
   }
 }
 
 resource "aws_route" "public" {
-  // Must pin count to avoid error "value of 'count' cannot be computed"
-  // See:
-  //   * <https://docs.cloudposse.com/troubleshooting/terraform-value-of-count-cannot-be-computed/>
-  //   * <https://github.com/hashicorp/terraform/issues/12570>
-  // Proper value once #12570 has some reasonable resolution:
-  //   count = "${var.create_peering_connection ? length(var.public_route_tables) : 0}"
-  count = "${var.create_peering_connection ? 1 : 0}"
-
-  route_table_id            = "${var.public_route_tables[count.index]}"
-  destination_cidr_block    = "${data.aws_vpc.peer.cidr_block}"
-  vpc_peering_connection_id = "${aws_vpc_peering_connection.this.id}"
+  count = var.create_peering_connection ? length(var.public_route_tables) : 0
```

Comment:
Reasonably sure this will still fail with some form of "count cannot be computed" if the user has one config that creates the public route tables and passes them to this module in the same config (this is our use case). Recommend setting up a test to validate.

Reply:
created test case `generated_route_tables` to address your use case and all tests still pass.